File: api-front/src/app/principal/Task.ts
export enum Status {
    FAZER = "FAZER",
    ANDAMENTO = "ANDAMENTO",
    FEITO = "FEITO"
}

export class Task {
    codigo: number = 0;
    nome: string = ''; // Propriedade existente para o nome do task
    status: Status = Status.FAZER; // Agora usando o enum Status como tipo
}

File: api/src/test/java/br/com/projeto/api/modelo/Task.java
package br.com.projeto.api.modelo;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name= "task")
@Getter
@Setter
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long codigo;

    private String nome;

    private Status status; // Campo de status que armazena a enumeração

    // Enumeração interna da classe Task
    public enum Status {
        FAZER,
        ANDAMENTO,
        FEITO
    }
}
